### PR TITLE
Fixed uncle pruning

### PR DESCRIPTION
**Pull request by arkpar**
I would also suggest uncle pruning to be moved from `on_initalize` to `set_uncles`

### CODE REVIEW

File: srml/authorship/src/lib.rs [hunk 1]
```diff
@@ -387,6 +372,18 @@ impl<T: Trait> Module<T> {
 		// check uncle validity.
 		T::FilterUncle::filter_uncle(&uncle, accumulator)
 	}
+
+	fn prune_old_uncles(minimum_height: T::BlockNumber) {
+		let mut uncles = <Self as Store>::Uncles::get();
```
Comment (tomusdrw) [inline srml/authorship/src/lib.rs]: We don't have a way to get an iterator without allocating the entire thing, right?

Comment (bkchr) [reply srml/authorship/src/lib.rs]: No

Comment (kianenigma) [reply srml/authorship/src/lib.rs]: Shouldn't we though (in the ling future probably)? 

sounds like a good enhancement: sth that only decodes elements of collection one by one in an iterator-like fashion and yields them.

Comment (bkchr) [reply srml/authorship/src/lib.rs]: What is the ling future? Some Chinese future? :D

My brain already started working on this task as @tomusdrw mentioned it :P

Comment (kianenigma) [reply srml/authorship/src/lib.rs]: > What is the ling future? Some Chinese future? :D

it comes from the same origin as _fail enough_ if you know what I am talking about 😁

> My brain already started working on this task as @tomusdrw mentioned it :P

Sweet.

Comment (rphmeier) [reply srml/authorship/src/lib.rs]: Theoretically, the storage APIs could let you choose which position to read from and you could read into a buffer. But at the end of the day you have to load the entire trie leaf, and that'll be a big one. Automatically splitting a vector over many storage entries and having convenient APIs for access would be useful

**Pull request merged**